pkg/cmd/create/create_devpod.go

```diff
@@ -158,6 +158,20 @@ func NewCmdCreateDevPod(commonOpts *opts.CommonOptions) *cobra.Command {
 	return cmd
 }
 
+// GetGitLabel try to reduce the label key and value size if they exceed 64 characters in length
+func GetGitLabel(gitInfo *gits.GitRepository, importURL string) (string, string) {
+	gitLabelKey := fmt.Sprintf("%s-%s-%s-%s", kube.LabelDevPodGitPrefix, gitInfo.Host, gitInfo.Organisation, gitInfo.Name)
+	gitLabelValue := naming.ToValidNameWithDots(importURL)
+
+	// try to minimize label size reducing git host to first part
+	if len(gitLabelKey) > 63 || len(gitLabelValue) > 63 {
```

Comment:
See above, if you use `ToValidNameTruncated` you get the name truncated when needed and the full name when the length is above 63 characters. In this case you don't have to do another check.